### PR TITLE
Pb-3: added category link  and product link selector

### PR DESCRIPTION
**Pull request by danyloherasymovscandiweb**

### CODE REVIEW

File: app/code/Scandiweb/DemoWidget/Block/Adminhtml/Widget/CategoryChooser.php [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+    {
+        if (!$this->getUseMassaction()) {
+            $chooserJsObject = $this->getId();
+            return '
+                function (grid, event) {
+                    var trElement = Event.findElement(event, "tr");
+                    var productId = trElement.down("td").innerHTML;
+                    var productName = trElement.down("td").next().next().innerHTML;
+                    var optionLabel = productName;
+                    var optionValue = "product/" + productId.replace(/^\s+|\s+$/g,"");
```
Comment (raivisdejus) [inline app/code/Scandiweb/DemoWidget/Block/Adminhtml/Widget/CategoryChooser.php]: If we are overwriting this, you can change this part so it does not add the extra `"product/"`
Product processing will be a bit simpler as you will not need to strip this part

Comment (raivisdejus) [reply app/code/Scandiweb/DemoWidget/Block/Adminhtml/Widget/CategoryChooser.php]: Also I am curious why the category has `procuct` stuff as prefix

File: app/code/Scandiweb/DemoWidget/Block/Adminhtml/Widget/Chooser.php [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+namespace Scandiweb\DemoWidget\Block\Adminhtml\Widget;
+
+use Magento\Backend\Block\Widget\Grid;
+use Magento\Backend\Block\Widget\Grid\Column;
+use Magento\Backend\Block\Widget\Grid\Extended;
+use Magento\Framework\Data\Form\Element\AbstractElement;
+use Magento\Catalog\Api\CategoryRepositoryInterface;
+use Magento\Framework\Data\Form\Element\Factory;
+
+class Chooser extends Extended
```
Comment (raivisdejus) [inline app/code/Scandiweb/DemoWidget/Block/Adminhtml/Widget/Chooser.php]: `Chooser` is a bad class name as there may be different choosers, best to name it more precisely e.g. `ProductChooser`

File: app/code/Scandiweb/DemoWidget/Block/Adminhtml/Widget/Chooser.php [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+    {
+        if (!$this->getUseMassaction()) {
+            $chooserJsObject = $this->getId();
+            return '
+                function (grid, event) {
+                    var trElement = Event.findElement(event, "tr");
+                    var productId = trElement.down("td").innerHTML;
+                    var productName = trElement.down("td").next().next().innerHTML;
+                    var optionLabel = productName;
+                    var optionValue = "product/" + productId.replace(/^\s+|\s+$/g,"");
```
Comment (raivisdejus) [inline app/code/Scandiweb/DemoWidget/Block/Adminhtml/Widget/Chooser.php]: also here, we can skip the `product/` part so it is easier for us to work with the data

File: app/code/Scandiweb/DemoWidget/Block/Adminhtml/Widget/CategoryChooser.php (hunk 1 above)
Comment (raivisdejus) [inline app/code/Scandiweb/DemoWidget/Block/Adminhtml/Widget/CategoryChooser.php]: Can we skip `product/` so it is easier for us to work with the data?

File: app/code/Scandiweb/DemoWidget/etc/widget.xml [hunk 4]
```diff
@@ -11,6 +12,9 @@
          'id' has to be unique for each widget.
     -->
     <widget id="DemoWidget" class="Scandiweb\DemoWidget\Block\Widget\Demo" placeholder_image="Scandiweb_DemoWidget::logo.png">
+=======
```
Comment (raivisdejus) [inline app/code/Scandiweb/DemoWidget/etc/widget.xml]: Merge conflict was not resolved here. Remove not needed parts, and commit your changes

File: app/code/Scandiweb/DemoWidget/etc/widget.xml [hunk 5]
```diff
@@ -55,8 +48,8 @@
                 <block class="Scandiweb\DemoWidget\Block\Adminhtml\Widget\ColorPicker">
                 </block>
             </parameter>
-            <parameter name="show_link" xsi:type="select" visible="true" sort_order="60">
```
Comment (raivisdejus) [inline app/code/Scandiweb/DemoWidget/etc/widget.xml:58]: Let's keep the `show_link` version, it is more descriptive than `is_enabled`

File: app/code/Scandiweb/DemoWidget/etc/widget.xml [hunk 6]
```diff
@@ -80,6 +73,27 @@
...
+                        <item name="button" xsi:type="array">
+                            <item name="open" xsi:type="string" translate="true">Select Product...</item>
+                        </item>
+                    </data>
+                </block>
+            </parameter>
+
+            <parameter name="category" xsi:type="block" visible="true" required="true" sort_order="110">
+                <label translate="true">Category</label>
+                <block class="Magento\Catalog\Block\Adminhtml\Product\Widget\Chooser">
```
Comment (raivisdejus) [inline app/code/Scandiweb/DemoWidget/etc/widget.xml]: Let's have this as `CategoryChooser`
Do we have one class for both product and category choosing?

File: scandipwa/packages/demo-widget/src/component/DemoWidget/DemoWidget.component.js [hunk 7]
```diff
@@ -1,138 +1,167 @@
+/* eslint-disable no-console */
```
Comment (raivisdejus) [inline scandipwa/packages/demo-widget/src/component/DemoWidget/DemoWidget.component.js]: let's remove this

File: scandipwa/packages/demo-widget/src/component/DemoWidget/DemoWidget.container.js [hunk 8]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-console */
```
Comment (raivisdejus) [inline scandipwa/packages/demo-widget/src/component/DemoWidget/DemoWidget.container.js]: let's remove this

**Comment by raivisdejus** [comment]
Please see previous commdents

File: app/code/Scandiweb/DemoWidget/etc/widget.xml [hunk 9]
```diff
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <widgets xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Widget:etc/widget.xsd">
+<<<<<<< HEAD
```
Comment (raivisdejus) [inline app/code/Scandiweb/DemoWidget/etc/widget.xml]: this is an indication of merge conflict

**Comment by danyloherasymovscandiweb** [comment]
Let's see if I will see the comments

**Review by raivisdejus — APPROVED** [review]

**Pull request merged**